napari/layers/utils/stack_utils.py

```diff
@@ -0,0 +1,171 @@
+import warnings
+from typing import Union, List, Tuple
+import itertools
+
+import numpy as np
+from napari.layers import Image, Labels
+from vispy.color import Colormap
+from ...utils import colormaps
+
+
+def stack_to_images(
+    stack: Union[Image, Labels],
+    axis: int,
+    colormap: List[Union[str, Colormap]] = None,
+    contrast_limits: List[List[int]] = None,
+    gamma: List[float] = None,
+    blending: str = None,
+    scale: Tuple[float] = None,
+    translate: Tuple[float] = None,
+) -> List[Image]:
+    """Function to split the active layer into separate layers along an axis
+
+    Parameters
+    ----------
+    stack : napari.layers.Image
+        The image stack to be split into a list of image layers
+    axis : int
+        The axis to split along.
+
+    Returns
+    -------
+    list
+        List of images
+    """
+
+    data = stack.data
+    name = stack.name
+    num_dim = len(data.shape)
+    n_channels = data.shape[axis]
+
+    if num_dim < 3:
+        warnings.warn(
+            "The image needs more than 2 dimensions for splitting", UserWarning
+        )
+        return None
+
+    if axis >= num_dim:
+        warnings.warn(
+            "Can't split along axis {}. The image has {} dimensions".format(
+                axis, num_dim
+            ),
+            UserWarning,
+        )
+        return None
+
+    if colormap is None:
+        if n_channels == 2:
+            colormap = iter(colormaps.MAGENTA_GREEN)
+        if n_channels > 2:
+            colormap = itertools.cycle(colormaps.CYMRGB)
+    else:
+        colormap = iter(colormap)
```

Comment:
I also think we should cycle here for times when you have more channels than colours:

```suggestion
        colormap = itertools.cycle(colormap)
```